lib/linter.js

```diff
@@ -1007,6 +1007,35 @@ class Linter {
     }
 
     /**
+     * Verifies the text against the rules specified by the second argument.
+     * @param {string|SourceCode} textOrSourceCode The text to parse or a SourceCode object.
+     * @param {ESLintConfig} config An ESLintConfig instance to configure everything.
+     * @param {(string|Object)} [filenameOrOptions] The optional filename of the file being checked.
+     *      If this is not set, the filename will default to '<input>' in the rule context. If
+     *      an object, then it has "filename", "saveState", and "allowInlineConfig" properties.
+     * @param {boolean} [saveState] Indicates if the state from the last run should be saved.
+     *      Mostly useful for testing purposes.
+     * @param {boolean} [filenameOrOptions.allowInlineConfig] Allow/disallow inline comments' ability to change config once it is set. Defaults to true if not supplied.
+     *      Useful if you want to validate JS without comments overriding rules.
+     * @param {function(string): string[]} [filenameOrOptions.preprocess] preprocessor for source text. If provided, this should
```

Comment:
Since this is a public API, this would make it a breaking change.

Reply:
Could you elaborate on why it would be a breaking change? It seems like this is just adding additional functionality, not breaking any existing functionality.

Reply:
Ah, sorry, for some reason I though it's a new argument, not a new property on the object. I'm not really sure how to treat those. On one hand it's a change of a signature of publicly available API, on the other - JS works without it just fine, so it's not a breaking change, I guess.

Reply:
I don't think there's any ambiguity here. Code that would previously work when using the API will still work, so it's not a breaking change. (This would also be the case if a new argument was added.)